p2p/client_test.go

```diff
@@ -112,6 +113,35 @@ func TestDiscovery(t *testing.T) {
 	assert.Contains(clients[4].host.Network().Peers(), clients[3].host.ID())
 }
 
+func TestGossiping(t *testing.T) {
+	assert := assert.New(t)
+	logger := &TestLogger{t}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	clients := startTestNetwork(ctx, t, 5, map[int][]int{
+		1: []int{0},
+		2: []int{0},
+		3: []int{1},
+		4: []int{2},
+	}, logger)
+
+	// wait for clients to finish refreshing routing tables
+	clients.WaitForDHT()
+
+	time.Sleep(5 * time.Second)
+
+	// gossip from client 4
+	err := clients[4].GossipTx(ctx, []byte("sample tx"))
```

Comment:
So the problem seems to be that the message we are trying to publish does not get validate properly: https://github.com/libp2p/go-libp2p-pubsub/blob/5457a2845b06d859df6d3683867808e46a15c500/validation.go#L242-L247